SwrveSDK/SDK/Talk/SwrveBaseCampaign.m

```diff
@@ -167,22 +168,21 @@ -(void)loadRulesFrom:(NSDictionary*)json
 -(void)loadTriggersFrom:(NSDictionary*)json
 {
     NSArray* jsonTriggers = [json objectForKey:@"triggers"];
+    
     if (!jsonTriggers) {
         DebugLog(@"Error loading triggers", nil);
         return;
     }
     
-    for (NSString* trigger in jsonTriggers){
+    for (NSDictionary* trigger in jsonTriggers){
         if (trigger) {
-            [self.triggers addObject:[trigger lowercaseString]];
+
+            SwrveTrigger *swrveTrigger = [[SwrveTrigger alloc] initWithDictionary:trigger];
+            if([swrveTrigger isValidTrigger]) {
```

Comment:
I don't think we need isValidTrigger. The logic should all be self contained in SwrveTrigger or some other class/manager thats holds the NSArray of valid swrvetriggers. You could also put the method canTriggerWithEvent in this other class. This makes it far easier to unit test as all logic is in one place.